src/services/world-state-store/client.ts

```diff
@@ -0,0 +1,95 @@
+import { Struct, type JsonValue } from '@bufbuild/protobuf';
+import type { CallOptions, Client } from '@connectrpc/connect';
+import { WorldStateStoreService } from '../../gen/service/worldstatestore/v1/world_state_store_connect';
+import {
+  GetTransformRequest,
+  ListUUIDsRequest,
+  StreamTransformChangesRequest,
+} from '../../gen/service/worldstatestore/v1/world_state_store_pb';
+import type { RobotClient } from '../../robot';
+import type { Options } from '../../types';
+import { doCommandFromClient } from '../../utils';
+import type { WorldStateStore } from './world-state-store';
+
+/**
+ * A gRPC-web client for a WorldStateStore service.
+ *
+ * @group Clients
+ */
+export class WorldStateStoreClient implements WorldStateStore {
+  private client: Client<typeof WorldStateStoreService>;
+  public readonly name: string;
+  private readonly options: Options;
+  public callOptions: CallOptions = { headers: {} as Record<string, string> };
+
+  constructor(client: RobotClient, name: string, options: Options = {}) {
+    this.client = client.createServiceClient(WorldStateStoreService);
+    this.name = name;
+    this.options = options;
+  }
+
+  async listUUIDs(extra = {}, callOptions = this.callOptions) {
```

Comment:
For all of these functions we should convert the bytes uuids to string before they hit the user. Maybe use something like this? https://github.com/domske/uuid-tool

Reply:
I updated to include a specific new type and utility function for handling this. Since we are just interacting with UUIDs here I wanted to keep it scoped pretty tightly, but we can always elevate it if we think this is useful for other resources.